packages/nx/src/hasher/hasher.ts

```diff
@@ -361,6 +361,48 @@ class TaskHasher {
     };
   }
 
+  private traverseExternalNodesDependencies(
+    projectName: string,
+    visited: string[]
+  ) {
+    const dependencies = this.projectGraph.dependencies[projectName] ?? [];
+    dependencies.forEach((d) => {
+      if (visited.indexOf(d.target) === -1) {
+        visited.push(d.target);
+        this.traverseExternalNodesDependencies(d.target, visited);
+      }
+    });
+  }
+
+  private hashTarget(projectName: string, targetName: string): PartialHash {
```

Comment:
it has to be the other way around. 

We don't know what your executor depends on. It's not like run-commands is an exception. It's the opposite: nrwl/webpack etc are exceptions

Reply:
It searches for executors in the `externalNodes` and if found returns the hash of that package and its dependencies. For `run-commands` we know we shouldn't look for the dependencies of `@nrwl/workspace`.

Are you saying that everything that isn't `@nrwl/*` should be considered as `hash-all-external-nodes`?

Reply:
Done!